### ISSUE TITLE
Video Player

### ISSUE THREAD

**Issue by inkeliz** [gioui-plugins/gio-plugins#25]
**Why that plugin should be added?**
Video/Music players are very common on websites and apps. I think multiple apps would like to play videos or even streaming videos from external source.

**What features this plugin should have and how it should work?**
It should work similar to "Webviewer", or better. The idea is to have `mediaplayer.RectOp{}` which defines the area to show the content and fill it with the video.

**What alternatives you've considered?**
In some OSes (desktops) it's possible to use `os.Open` and `os.Create`, but that doesn't opens the file-dialog. So, that requires to create a custom file-dialog in-gio.

**Is this plugin implementable on all OSes (Android, iOS, macOS, WASM, Windows...)?**
No.

**Did you known any particular limitation for specific OS?**
On WASM it's not possible to directly access video codecs, it would require WebCodecs API, which is only supported by Chromium. It could fallback to `<video>`, but makes customization harder. Also, the video-codecs depends on the system itself, not only OS. It's possible to have one Windows machine without AV1, and another one with AV1 codec. On Andriod/iOS/macOS that is more difficult. 

Handling the video-codec itself (without external system dependency) is VERY hard, and also lacks hardware-acceleration (such as GPU).

**Comment by gedw99** [comment]
Yeah i also think a webview is the way to go. Let the browser do the work.

Do you know if the gio webview can be placed within a layout yet ?

**Comment by gedw99** [comment]
Btw thanks for raising this one. Have been hitting my head against brick wall on video :)

**Comment by ddkwork** [comment]
I'll take the time to look into soon

**Comment by gedw99** [comment]
Yeah I agree.  FFmpeg is just the wrong fit. Fine for a server but not on a client.

Regarding native OS codec, for example, for IOS and Mac there is https://github.com/progrium/macdriver/tree/main/macos/avfoundation. It might be a decent reference, but I have not tried this.

Any API we use as you say needs break down the video into image frames to feed to GIO. Same with Audio

There is a GIO video and audio player at least.  
https://github.com/mearaj/gio-video. Nice stuff @mearaj

The other aspect of accessing the native Audi and Video API will be that it will also make it easy to do a Camera plugs and to display the Video and Audio with the same gio video player that @ mearaj has started off.

**Comment by gedw99** [comment]
Hey

@inkeliz 
@mearaj

I have not tried it yet, but https://github.com/arthenica/ffmpeg-kit looks like it wil give us what we want with regarding to video on Mobile and Desktop.
